src/test/java/org/mockito/internal/util/reflection/GenericTypeExtractorTest.java

```diff
@@ -64,5 +65,7 @@
         assertEquals(Integer.class, genericTypeOf(IDeeper.class, Base.class, IBase.class));
         assertEquals(Integer.class, genericTypeOf(Crazy.class, Base.class, IBase.class));
         assertEquals(Integer.class, genericTypeOf(Crazier.class, Base.class, IBase.class));
+
+        assertEquals(Integer.class, genericTypeOf(SecondGeneric.class, Base.class, IBase.class));
```

Comment:
These should have been separate test methods, but this is consistent. Feel free to send a follow-up PR to fix that.